### ISSUE TITLE
- WARNING - Interrompendo treinamento após 30 minutos

### ISSUE THREAD

**Issue by SyraDevOps** [SyraDevOps/Luna#6]
{'loss': 0.2022, 'grad_norm': 1.5289089679718018, 'learning_rate': 3.177888888888889e-05, 'epoch': 72.89}
{'loss': 0.2015, 'grad_norm': 0.9576975703239441, 'learning_rate': 3.176777777777778e-05, 'epoch': 72.93}
 36% 16418/45000 [29:59<50:32,  9.43it/s]WARNING:src.utils.callbacks:Interrompendo treinamento após 30 minutos
2025-07-08 19:47:19,559 - src.utils.callbacks - WARNING - Interrompendo treinamento após 30 minutos
 36% 16419/45000 [30:00<50:18,  9.47it/s]INFO:src.utils.callbacks:Concluída época 72.97333333333333/200
2025-07-08 19:47:19,559 - src.utils.callbacks - INFO - Concluída época 72.97333333333333/200

# Retirar timeout

**Comment by SyraDevOps** [comment]
I'll scan the workspace to find where the timeout is enforced and how the callback is instantiated so we can safely disable it without breaking other logic. Entendi: você quer remover o timeout que está interrompendo o treinamento após 30 minutos. Vou desabilitar essa lógica no callback e ajustar a criação do Trainer para não forçar o timeout.

Checklist
- Tornar o timeout opcional no `CustomTrainingCallback` em callbacks.py (desabilitado por padrão).
- Ajustar trainer.py para não passar `timeout_min=30` (ou passar None).
- Deixar claro como validar que o treinamento não para mais por tempo.

## Mudanças propostas

### 1) Desabilitar o timeout no callback (timeout opcional, desativado por padrão)

- Altera o `timeout_min` para `None` por padrão.
- Só calcula `self.timeout_time` quando `timeout_min` for um número positivo.
- Garante que checagens de timeout só rodem quando `self.timeout_time` existir.

````python
from transformers import TrainerCallback, TrainerState, TrainerControl
from datetime import datetime, timedelta
# ...existing code...
logger = logging.getLogger(__name__)

class CustomTrainingCallback(TrainerCallback):
    """Callback personalizado para monitorar treinamento"""
    
    def __init__(self, timeout_min=None, monitor_memory=False):
        """Inicializa o callback com um timeout em minutos (None desativa o timeout)"""
        self.timeout_min = timeout_min  # None => sem timeout
        self.start_time = None
        self.timeout_time = None
        self.last_log_time = None
        self.log_interval = 60  # Logar a cada 60 segundos
        self.monitor_memory = monitor_memory
        self.peak_memory_usage = 0
        
    def on_train_begin(self, args, state, control, **kwargs):
        """Chamado quando o treinamento começa"""
        self.start_time = time.time()
        if self.timeout_min is not None and self.timeout_min > 0:
            self.timeout_time = self.start_time + (self.timeout_min * 60)
        else:
            self.timeout_time = None
        self.last_log_time = self.start_time
        # ...existing code...
    
    def on_step_end(self, args, state, control, **kwargs):
        # ...existing code...
        # Interromper somente se um timeout válido foi configurado
        if self.timeout_time is not None and time.time() >= self.timeout_time:
            logger.warning(f"Interrompendo treinamento após {self.timeout_min} minutos")
            control.should_training_stop = True
        # ...existing code...

    def on_log(self, args, state, control, logs=None, **kwargs):
        # ...existing code...
        # Se houver qualquer checagem de tempo aqui, também proteger:
        # if self.timeout_time is not None and time.time() >= self.timeout_time:
        #     ...
        # ...existing code...
````

Observação: Se houver outras checagens de tempo em `on_log` ou `on_epoch_*`, proteja-as com `if self.timeout_time is not None ...` da mesma forma (o trecho acima já indica como fazer). Isso mantém o callback útil (logs, monitoramento de memória), mas sem parar por tempo quando `timeout_min` estiver desativado.

### 2) Não forçar timeout na criação do Trainer

Trocar a passagem fixa de 30 min por None (ou simplesmente não passar o callback, caso não precise de logs/monitoramento). Como você ainda parece querer os logs, vamos manter o callback com timeout desativado.

````python
# ...existing code...
from src.utils.callbacks import CustomTrainingCallback
# ...existing code...

# Dentro da criação do Trainer (linha ~189, conforme grep):
trainer = Trainer(
    # ...existing args...
    callbacks=[CustomTrainingCallback(timeout_min=None)]
)
# Alternativa: remover completamente o parâmetro callbacks se não precisar do callback:
# trainer = Trainer(
#     # ...existing args...
# )
````

Se você preferir não tocar no trainer, basta manter o callback e mudar o default em callbacks.py para `None` (como no passo 1). Ainda assim recomendo atualizar a linha do trainer.py para explicitar a intenção de não usar timeout.

## Como validar

- Reexecute um treinamento curto e observe que a mensagem de timeout não aparece mais e o treino não é interrompido por tempo.
- Opcionalmente, rode os testes (se configurados):

```powershell
python -m pytest -q
```

- Se quiser manter logs de recursos sem timeout, você pode ativar o monitoramento de memória:

```python
callbacks=[CustomTrainingCallback(timeout_min=None, monitor_memory=True)]
```

## Cobertura dos requisitos
- Remover timeout que interrompe o treino: Feito (timeout desativado por padrão e guardas adicionadas; trainer atualizado para não forçar 30 min).
- Manter funcionamento normal do treinamento: Mantido (callback segue registrando logs e sem interrupção por tempo).
- Instruções de validação: Incluídas.

Se quiser, posso também alinhar a versão do `CustomTrainingCallback` duplicada em callbacks.py para evitar confusão futura, mas hoje ela não é usada pelo `trainer`.
